UnitystationLauncher/ViewModels/SecurityPanelViewModel.cs

```diff
@@ -0,0 +1,538 @@
+
+                    string response = await msgBox.Show();
+
+                    await _writer.WriteLineAsync(response == "No" ? false.ToString() : true.ToString());
+                    await _writer.FlushAsync();
+                    return Task.CompletedTask;
+                });
+            }
+        }
+    }
+
+    public void Dispose()
+    {
+        _serverPipe.Dispose();
+        _reader?.Dispose();
+        _writer?.Dispose();
+    }
+}
+
+public class SecurityPanelViewModel : PanelBase
+{
+    public override string Name => "Security";
+    public override bool IsEnabled => false;
+
+    private readonly IAssemblyChecker _IAssemblyChecker;
+
+    public SecurityPanelViewModel(IAssemblyChecker assemblyChecker)
+    {
+        _IAssemblyChecker = assemblyChecker;
+    }
+
+
+    public override void Refresh()
+    {
+        //what does this do?!?
+    }
+
+    //TODO 
+    //Automatic start-up pipe
+    //Zip bomb prevention
+
+
+    //TODO
+    // in-memory virtual file system
+
+    //UnityPlayer.dll?? look in to
+    //Unitystation.exe
+
+    //Cleanup files end 
+
+    // if (sandboxConfig.WhitelistedAssembliesDEBUG.Contains(
+    //(type.ResolutionScope as AssemblyTypeChecker.MResScopeAssembly).Name))
+
+    // if (string.Equals(fileName, assemblyName.Name, StringComparison.OrdinalIgnoreCase))  Potential security+
+
+    //Check the multi-assembly stuff, make sure you can't do any naughty with name matching and stuff
+    //tool to Make a PR Auto to add Two white lists, if PR violates White list
+
+    //TODO Double check added stuff
+    // UnityEngine.Events
+    // UnityEngine.Events.UnityAction TODO Looking 
+    //"UnityEngine.UI":{Dropdown"
+    // Collider2D
+    // LayerMask
+    // Physics2D
+    // Mathf???
+    // UnityEngine.Transform???
+    // Random???
+    // Time
+    // UnityEvent
+    // UnityEngine.Plane?
+    // UnityEngine.Rect?
+    // Ray?
+    // Texture2D
+    //Manually set 
+    //Texture2D
+    //ParticleSystem
+    //Input
+    //GUIUtility
+    //GUI?
+    //ImageConversion 
+    //Application
+    //UnityEngine.Resources
+    //Addressables
+
+
+    //TODO Patch
+    //UnityEngine.Events.UnityEvent, GetValidMethodInfo
+
+
+    public void OnSetUpPipe()
+    {
+        var data = new PipeHubBuildCommunication();
+        new Task(data.StartServerPipe).Start();
+    }
+
+    public void OnUpdateSafe()
+    {
+        DirectoryInfo directory = new DirectoryInfo(System.AppDomain.CurrentDomain.BaseDirectory);
+
+        DirectoryInfo stagingDirectory = directory.CreateSubdirectory("staging");
+
+        stagingDirectory = stagingDirectory.CreateSubdirectory("unitystation_Data");
+        stagingDirectory = stagingDirectory.CreateSubdirectory("Managed");
+
+        DirectoryInfo goodDirectory = directory.CreateSubdirectory("GoodFiles");
+        goodDirectory = goodDirectory.CreateSubdirectory("VDev");
+        goodDirectory = goodDirectory.CreateSubdirectory("Managed");
```

Comment:
I agree with Codacy here, I think these folder names should be constant values to reduce the risk of typos when this is edited in the future.

Reply:
yeah,  I'm in the process of Converting it to a new system do that as part of it